src/dom/dimensions-2d-dom.ts

```diff
@@ -0,0 +1,21 @@
+export class Dimensions2dDom {
+  static fromElementOffset(element: HTMLElement) {
+    return new Dimensions2dDom(element.offsetWidth, element.offsetHeight);
+  }
+  width: number;
+  height: number;
+
+  constructor(width: number, height: number) {
+    this.width = width;
+    this.height = height;
+  }
+
+  equals(candidate: Dimensions2dDom) {
+    return this.width === candidate.width &&
+        this.height === candidate.height;
+  }
+
+  getHypotenuseLength() {
+    return Math.sqrt(Math.pow(this.width, 2) + Math.pow(this.height, 2));
```

Comment:
`this.width * this.width` will be significantly faster than `Math.pow(this.width, 2)`

https://www.measurethat.net/Benchmarks/Show/1651/0/math-pow-vs-multiply

Reply:
Nice. Done :)